ts/session/utils/RingBuffer.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * This ringbuffer class can be used to keep a list of at most a size and removing old items first when the size is exceeded.
```

Comment:
Was readying this https://redisson.org/glossary/ring-buffer.html and it mentions using pointers for the current item in the array instead of shifting the entire list of array items. Do we care enough to implement this correctly?

Reply:
amended

Reply:
made it use indexes so we avoid slicing the array on each overflow

Reply:
nice